### PR TITLE
Add purge flag for message deletes

### PR DESCRIPTION
**Pull request by ericnewcomer**
This also fixes the test for `bulk_purge` which isn't in use in production (it was testing against a cached value). The changes for this PR made that test pass for real.

### CODE REVIEW

**Comment by nicpottier** [comment]
Taking over this PR to add the is_archived to counts

**Comment by nicpottier** [comment]
@ericnewcomer think this is ready to review now. We'll need one more PR after this to actually delete messages with a visibility of D (and remove filtering by it / constant) but this takes care of all the triggers etc.. (which won't need to change)

File: temba/msgs/tests.py [hunk 1]
```diff
@@ -1360,6 +1377,178 @@ def setUp(self):
         # a Twitter channel
         self.twitter = Channel.create(self.org, self.user, None, "TT")
 
+    def test_message_archiving(self):
+        test_cases = [
```
Comment (nicpottier) [inline temba/msgs/tests.py]: In the spirit of go!

File: temba/orgs/migrations/0040_auto_20180529_1951.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    dependencies = [("orgs", "0039_auto_20180202_1234")]
+
+    operations = [
+        migrations.RemoveField(model_name="debit", name="is_squashed"),
+        migrations.AlterField(
+            model_name="debit",
+            name="debit_type",
+            field=models.CharField(choices=[("A", "Allocation")], help_text="What caused this debit", max_length=1),
+        ),
+        migrations.RunPython(delete_purge_debits),
```
Comment (ericnewcomer) [inline temba/orgs/migrations/0040_auto_20180529_1951.py]: does django care about the order here?

**Pull request merged**